src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -618,6 +619,13 @@ bool primitive_inst::update_impl() {
                             cache.add(updated_params_no_dyn_pad, impl->clone());
                         }
                     });
+                } else {
+                    auto impl = _node->type()->choose_impl(*_node, updated_params_no_dyn_pad);
+                    if (!can_be_optimized()) {
+                        auto kernels = prog->get_kernels_cache().compile(updated_params_no_dyn_pad, impl->get_kernels_source());
+                        impl->set_kernels(kernels);
+                        cache.add(updated_params_no_dyn_pad, impl->clone());
+                    }
```

Comment:
I don't get the point of these changes. Async compilation is used only if we have shape agnostic impl for current node. It means that if async compilation is disabled for some reason, shape agnostic kernel must be used. 
"Sync" compilation is supposed to happen as a fallback for primitives w/o dynamic impl (and it's covered below)

Reply:
Removed this change